src/test/java/tests/junit/Tests.java

```diff
@@ -0,0 +1,195 @@
+package tests.junit;
+
+import dto.ItemPair;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+import utils.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static utils.Constants.FILE_NAME_LIST;
+
+public class Tests {
+
+
+    private static JsonParser parser = new JsonParser();
+    private static List<Cart> carts;
+    private static final List<ItemPair> itemPairs = Arrays.asList(
+            new ItemPair(FILE_NAME_LIST.get(0), 1, 1),
+            new ItemPair(FILE_NAME_LIST.get(1), 0, 2),
+            new ItemPair(FILE_NAME_LIST.get(2), 0, 3),
+            new ItemPair(FILE_NAME_LIST.get(3), 4, 0),
+            new ItemPair(FILE_NAME_LIST.get(4), 5, 0),
+            new ItemPair(FILE_NAME_LIST.get(5), 0, 0)
+    );
+
+    @BeforeAll
+    public static void setUp() {
+        carts = Utils.generateCart(itemPairs);
+        for (Cart cart : carts) {
+            parser.writeToFile(cart);
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {6})
```

Comment:
this is not the solution that I expect, several options 1 - use itemParts.size() - but it is a bad solution. 2 - use method source and "execute" this method exactly the same time as list capacity.

Reply:
fixed